Source/NujasCombat/Private/DynamicTargetingComponent.cpp

```diff
@@ -118,63 +120,79 @@ void UDynamicTargetingComponent::UpdateCameraLock()
-			}
-			else
-			{
-				LastActorId = ActorID;
-			}
+			maxLength = CamOffset.Size();
+			bCalculatedOffsetOnce = true;
 		}
-		else
+		else 
 		{
-			ActorHorizontalMovementMap.Add(ActorID, FHorizontalActorMovementData(ActorScreenSpacePos.X));
-			LastActorId = ActorID;
+			CamOffset = UKismetMathLibrary::ClampVectorSize(CamOffset, maxLength, maxLength);
 		}
-	}
+		CamOffset.Z = 0;
+		FVector FinalTarget = EnemyLocation + CamOffset;
+
+		// TODO: for a future PR
+		//EnemyLocation = TargetableActor->GetActorLocation();
+		//FVector EnemyVelocity = TargetableActor->GetVelocity();
+		//EnemyVelocity.Normalize();
+		//CamOffset += EnemyVelocity * 10.f; // + closeness * 5.f;
+
+		CamLocation.Z = FinalTarget.Z;
+		DrawDebugLine(GetWorld(), CamLocation, FinalTarget, FColor::Red, false, .01f, 0, 5.f);
+		DrawDebugLine(GetWorld(), EnemyLocation, FinalTarget, FColor::Blue, false, .01f, 0, 5.f);
+		DrawDebugLine(GetWorld(), CamLocation, EnemyLocation, FColor::Blue, false, .01f, 0, 5.f);
+		CamRotationToTarget = UKismetMathLibrary::FindLookAtRotation(CamLocation, FinalTarget); // change to plus
+		
+		/* TODO for a future PR
+		if(angle2 >= 0) 
+		{
+			DrawDebugLine(GetWorld(), EnemyLocation, EnemyLocation - CamOffset, FColor::Red, false, .1f, 0, 5.f);
+			// DrawDebugLine(GetWorld(), RightVec + CamLocation, DirFromCamToEnemy + CamLocation, FColor::Blue, true, .5f, 0, 5.f);
+			CamRotationToTarget = UKismetMathLibrary::FindLookAtRotation(CamLocation,  EnemyLocation - CamOffset); // change to plus
+		}
+		else 
+		{
+			DrawDebugLine(GetWorld(), EnemyLocation, EnemyLocation + CamOffset, FColor::Red, false, .1f, 0, 5.f);
+			// DrawDebugLine(GetWorld(), RightVec + CamLocation, DirFromCamToEnemy + CamLocation, FColor::Blue, true, .5f, 0, 5.f);
+			CamRotationToTarget = UKismetMathLibrary::FindLookAtRotation(CamLocation, EnemyLocation + CamOffset); // change to plus
+		}
+		*/
+		
+		
+		// bIterated = true;
+	// }
+	// if(!bIterated)
+		// return;
 
-	// instead of taking every character into account, take only one based on priority
-	// How to determine the most important enemy
-	// how far away is it from the player
-	// how far away is it from the center of the screen
-	// 
-	// The camera should ignore an enemy unless the new position is moving away from the center
-	// 
-	// OR
-	//
-	// Collect the average screen space positions and interpolate towards it
-
-	// apply the summarized rotation to the yaw of the player via interpolation
 	FRotator ControllerRotation = PlayerController->GetControlRotation();
-	ControllerRotation.Yaw = FMath::FInterpTo(ControllerRotation.Yaw, ControllerRotation.Yaw + SummarizedSpeed, FApp::GetDeltaTime(), 13.5f);
-	PlayerController->SetControlRotation(ControllerRotation);
+	CamRotationToTarget.Pitch = ControllerRotation.Pitch;
+	CamRotationToTarget.Roll = ControllerRotation.Roll;
+	PlayerController->SetControlRotation(UKismetMathLibrary::RInterpTo(ControllerRotation, CamRotationToTarget, FApp::GetDeltaTime(), 150.f));
```

Comment:
Please use Fmath (Core) instead of KismetMath (BPLib). They are essentially the same:

https://docs.unrealengine.com/en-US/API/Runtime/Core/Math/FMath/RInterpTo/index.html

https://docs.unrealengine.com/en-US/API/Runtime/Engine/Kismet/UKismetMathLibrary/RInterpTo/index.html

Reply:
Also please move the interpolation into a variable before the call, make code easier to follow.

Reply:
Substituted the Kismet functions as well as simplified line 195